chainforge/react-server/src/backend/ai.ts

```diff
@@ -21,7 +21,14 @@ export class AIError extends Error {
 export type Row = string;
 
 // LLM to use for AI features.
-const LLM = "gpt-3.5-turbo";
+export const getLLM = function (apiKeys: Dict): string {
```

Comment:
If this is exported, can this be more descriptive than `getLLM`? 

I like what you are doing here with auto-detecting what models the user has access to; however, I also think this is a setting that should be added to the GlobalSettingsModal in the AI Support tab. There should be a switcher as a `NativeSelect` element that provides control over (and transparency about) what provider is being used for the genAI features.

Reply:
I thought about that but I wanted to be as least intrusive as possible with the change. I'll add it as a setting and submit for your review.

Reply:
Settings:

![image](https://github.com/ianarawjo/ChainForge/assets/38036163/f0e6560f-1a4c-48f9-b1ec-bcadcc11fad8)

Error message:

![image](https://github.com/ianarawjo/ChainForge/assets/38036163/8c21b53b-3f8a-42b1-9509-44a416a7b442)

No model is set by default so that the user has to make an explicit choice. This is the error the user gets:

![image](https://github.com/ianarawjo/ChainForge/assets/38036163/45b3d9e0-732e-4f68-9d1c-dce6e16a9e3a)

Reply:
> here should be a switcher as a NativeSelect element that provides control over (and transparency about) what provider is being used for the genAI features.

> So, for the "no model selected" I would change it to something like, "You have not entered any API keys supported by the GenAI feature. See Settings > AI Support for more details." I wouldn't make it an explicit selection.

To clarify: you would like to have a Provider selection in the Global Settings instead of an explicit model selection. Do you you want to have a default provider (in which case there is no need for a "No provider selected" warning, or do you want to have the user explicitly selecting the provider to use?

Regarding the AI features you mention, I would not like not knowing which model is used for what due to the significant difference in cost.

Reply:
Ah I see. I just don't want the user to have to pick a provider every time they go to the application. If you can make it so the user preference is saved, and they are only asked once to go to the Global Settings window to set the model to use for GenAI features, then I think that is a fair trade-off and choice. You can save it as a flag in the browser's localStorage to be simplest, but could also consier storing it as a config file when CF is running locally.

Reply:
With regards to what appears in the AI Support dropdown, technically we may call GPT-4 in the future (actually I think the evaluator generating code uses gpt-4 already). So, this should be more about the *family* of models and provider that will be queried. You could put Claude (Bedrock), then, where genAI calls an LLM, the function that fetches what LLM to use might have a "weak" and "strong" variant (your getLLM call would be slightly more complex there). I'm not sure what new Claude model is more powerful.